src/coder/test-utils.ts

```diff
@@ -4,8 +4,16 @@ import { DNSPacket } from "./DNSPacket";
 import { Question } from "./Question";
 import { ResourceRecord } from "./ResourceRecord";
 
+// Utility function to convert IPv4-mapped IPv6 addresses to IPv4
+function convertIPv4MappedIPv6ToIPv4(address: string): string {
```

Comment:
Suggested change...we just need to remove the prefix here since we've already got the address in the predicate.

```
function convertIPv4MappedIPv6ToIPv4(address: string): string  {
  return address.replace(/^::ffff:/i, "");
}
```